components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1745,6 +1745,16 @@ where
         let stepped = Cell::new(false);
         let memtrace_raft_entries = &mut self.fsm.peer.memtrace_raft_entries as *mut usize;
         defer!({
+            let do_check = || {
+                fail_point!("memtrace_raft_messages_overflow_check_peer_recv", |_| {
+                    true
+                });
+                false
+            };
+            if do_check() {
+                let prev = MEMTRACE_RAFT_MESSAGES.sum();
+                assert!(prev >= heap_size);
```

Comment:
Better to replace this with `debug_assert!`

Reply:
We need the send msg failpoint to trigger this assert.